README.md

```diff
@@ -75,6 +75,39 @@ If you use `MESH_LOWPAN_ND` or `MESH_THREAD` you need to specify your radio modu
     }
 ```
 
+If you use `CELLULAR_ONBOARD` you must specify the following:
+
+```json
+    "target_overrides": {
+        "*": {
+            "ppp-cell-iface.apn-lookup": true
```

Comment:
Can't we always set this? That would save user editing work.

Reply:
I'd be happy to, @hasnainvirk?

Reply:
@RobMeades I don't mind that but it changes the existing behaviour which is probably not intended. Check the comments on connect() api

```
    /** Attempt to connect to the Cellular network
     *
     *  Brings up the network interface. Connects to the Cellular Radio
     *  network and then brings up the underlying network stack to be used
     *  by the cellular modem over PPP interface.
     *
     *  If the SIM requires a PIN, and it is not set/invalid, NSAPI_ERROR_AUTH_ERROR is returned.
     *  For APN setup, default behaviour is to use 'internet' as APN string and assuming no authentication
     *  is required, i.e., username and password are not set. Optionally, a database lookup can be requested
     *  by turning on the APN database lookup feature. In order to do so, add 'MBED_CONF_PPP_CELL_IFACE_APN_LOOKUP'
     *  in your mbed_app.json. APN database is by no means exhaustive. It contains a short list of some public
     *  APNs with publicly available usernames and passwords (if required) in some particular countries only.
     *  Lookup is done using IMSI (International mobile subscriber identifier).
     *  Please note that even if 'MBED_CONF_PPP_CELL_IFACE_APN_LOOKUP' config option is set, 'set_credentials()' api still
     *  gets the precedence. If the aforementioned API is not used and the config option is set but no match is found in
     *  the lookup table then the driver tries to resort to default APN settings.
```

Reply:
Since none of us think it's a bad idea, and the mechanism will, in any case, drop back to using "internet" as the APN if any of the known APNs don't work, _and_ the Cellular API hasn't been out there long enough for many people to become fixed in their usage, I suggest that I modify the `mbed_lib.json` to set it to `true` and also modify the description to match

Reply:
Or, maybe, we leave this PR as it is and do that in a subsequent PR, just to avoid mixing the two up.

Reply:
I like the idea of a separate PR

Reply:
A suggestion, it would be nice to add a link to mbed-os cellular  documentation in Readme.

Reply:
Good idea, if you can point me at the right link I will re-submit this PR with it added.

Reply:
Here is the link to cellular documentation https://docs.mbed.com/docs/mbed-os-api-reference/en/latest/APIs/communication/cellular/

Reply:
Thanks, PR updated.